### PR TITLE
Rewrite and re-scope deployment documentation

### PR DESCRIPTION
**Pull request by ironcladlou**
Make the deployment doc less of a low level technical document/proposal
mashup, and rewrite it for end-user consumption. Architecture and
technical details can be moved to a new document. The upstream proposal
will also be a new document.

### CODE REVIEW

**Comment by ironcladlou** [comment]
This is a follow-on to https://github.com/openshift/origin/pull/584

**Comment by ironcladlou** [comment]
cc @pmorie @bparees @pweil-

File: docs/deployments.md [hunk 1]
```diff
@@ -1,306 +1,172 @@
...
+    "containerNames": [
+      "helloworld"
+    ],
+    "repositoryName": "openshift/origin-ruby-sample",
+    "tag": "latest"
+  }
+}
+```
+
+In this example, when the `latest` tag value for the `imageRepository` named `openshift/origin-ruby-sample` changes, every container in the `deploymentConfig` matching those in `containerNames` will be updated with the new tag value, and a new `deployment` will be created.
```
Comment (pweil-) [inline docs/deployments.md]: "every container in the `deploymentConfig` matching those in `containerNames`" reads funny to me.  

maybe: the containers specified in `containerNames` for `deploymentConfig` D will be updated ...

Comment (ironcladlou) [inline docs/deployments.md]: Incorporated, thanks

**Comment by pmorie** [comment]
LGTM [merge]

**Comment by pmorie** [comment]
[merge] looks like jenkins timeout

**Pull request merged**